perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1899,6 +1899,40 @@ public Application getApplicationById(PerunSession sess, int appId) throws Regis
 
 	}
 
+	@Override
+	public List<Application> getApplicationsForVo(PerunSession userSession, Vo vo, List<String> state, String dateFrom, String dateTo) throws PerunException {
+		vosManager.checkVoExists(userSession, vo);
+
+		//Authorization
+		if (!AuthzResolver.authorizedInternal(userSession, "getApplicationsForVo_Vo_List<String>_String_String_policy", Collections.singletonList(vo))) {
+			throw new PrivilegeException(userSession, "getApplicationsForVo");
+		}
+		try {
+			MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
+			StringBuilder query = new StringBuilder(APP_SELECT);
+			query.append(" where a.vo_id=:voId");
+			sqlParameterSource.addValue("voId", vo.getId());
```

Comment:
I think that when you specify only the VO, it will also return all group applications within that VO. I know that this is how also the old method works but I am not sure if it is correct. What do you think @zlamalp ?

Reply:
@balcirakpeter you are right. Original method returns also group applications within the VO on purpose. So the new method should do the same as its expected to be used from the same part of GUI. It was probably just not reflected in javadoc.

@mvocu we should add some kind of boolean flag in API of those methods (defaulting to old behaviour), since I see we use it from ``AppAutoRejectionScheduler``, where it should process only VO applications, but like this it works also on group applications. For now it doesn't have to be in outer API (openAPI/RPC). Its just for the purpose of AppAutoRejectionScheduler.

Reply:
OK, parameter added, references fixed.